MediaStore/app/src/main/java/com/android/samples/mediastore/MainActivity.kt

```diff
@@ -156,39 +187,66 @@ class MainActivity : AppCompatActivity() {
      */
     private fun requestPermission() {
         if (!haveStoragePermission()) {
-            val permissions = arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE)
+            val permissions = arrayOf(
+                Manifest.permission.READ_EXTERNAL_STORAGE,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE
+            )
             ActivityCompat.requestPermissions(this, permissions, READ_EXTERNAL_STORAGE_REQUEST)
         }
     }
 
+    private fun deleteImage(image: MediaStoreImage) {
+        MaterialAlertDialogBuilder(this)
+            .setTitle(R.string.delete_title)
+            .setMessage(getString(R.string.delete_message, image.displayName))
+            .setPositiveButton(R.string.delete_positive) { _: DialogInterface, _: Int ->
+                viewModel.deleteImage(image)
+            }
+            .setNegativeButton(R.string.delete_negative)
+            { _: DialogInterface, _: Int ->
+                // Nothing to do
```

Comment:
Usually you'd explicitly call `dialog.dismiss()`

Reply:
Done.